moped-editor/src/utils/dateAndTime.js

```diff
@@ -57,11 +57,58 @@ export const makeHourAndMinutesFromTimeStampTZ = timeStampTZString => {
  * @param {String} timeStampTZString - Timestamptz string in ISO format
  * @return {String} Time formatted as HH:MM:SS XM
  */
-export const makeFullTimeFromTimeStampTZ = timeStampTZString => {
+export const makeFullTimeFromTimeStampTZ = (timeStampTZString) => {
   const dateObject = new Date(timeStampTZString);
   return dateObject.toLocaleTimeString("en-US", {
     hour: "numeric",
     minute: "numeric",
     second: "numeric",
   });
 };
+
+const MILLS_CONVERSION = {
+  minute: 60_000,
+  hour: 3_600_000,
+  day: 86_400_000,
+};
+
+/**
+ * Format millseconds into a relative description
+ * @param {integer} millsElapsed - the elapsed time in milliseconds to format
+ * @param {string} unitName - the name of the units to convert to - the string must
+ * be defined in the MILLS_CONVERSION objeect
+ * @returns {string} in format  `<value> <unit-name> ago`
+ */
+const formatRelativeTime = (millsElapsed, unitName) => {
+  const factor = MILLS_CONVERSION[unitName];
+  const millsConverted = Math.round(millsElapsed / factor);
+  const singular = millsConverted === 1;
+  return `${millsConverted} ${unitName}${singular ? "" : "s"} ago`;
+};
+
+/**
+ * Returns a humanized duration of a timestamp relative to now
+ * Behaves similarly to https://github.com/github/relative-time-element
+ * @param {*} timeStampTZString - Timestamp string in ISO format
+ * @returns {string} a humanized description of a timestamp as follows:
+ * - less than 1 minute: 'Just now'
+ * - less than 1 hour: 'X minute(s) ago'
+ * - less than 1 day: 'x hours ago'
+ * - less than 30 days: 'x days ago'
+ * - else: Date.toLocaleTimeString() (e.g., 1/23/2023)
+ */
+export const formatRelativeDate = (timeStampTZString) => {
+  const targetDate = new Date(timeStampTZString);
+  const millsElapsed = new Date().getTime() - targetDate.getTime();
+  if (millsElapsed > MILLS_CONVERSION.day * 30) {
+    return targetDate.toLocaleDateString("en-US");
+  } else if (millsElapsed < MILLS_CONVERSION.minute) {
+    return "Just now";
+  } else if (millsElapsed < MILLS_CONVERSION.hour - MILLS_CONVERSION.minute) {
+    return formatRelativeTime(millsElapsed, "minute");
+  } else if (millsElapsed < MILLS_CONVERSION.day - MILLS_CONVERSION.hour) {
```

Comment:
we subtract one notch from the next smaller unit of measure so that later on so that 59 minutes and 30 seconds return "1 hour" instead of "60 minutes".